packages/cactus-plugin-ledger-connector-iroha/src/main/typescript/api-client/iroha-api-client.ts

```diff
@@ -0,0 +1,247 @@
+      this.log.error("Error (connect_timeout): ", err);
+      socket.disconnect();
+      subject.error(err);
+    });
+
+    return subject.pipe(
+      finalize(() => {
+        this.log.info("FINALIZE - unsubscribing from the stream...");
+        socket.emit(IrohaSocketSessionEvent.Unsubscribe);
+        socket.disconnect();
+      }),
+      share(),
+    );
+  }
+
+  /**
+   * Immediately sends request to the validator, doesn't report any error or responses.
+   * @param args - arguments.
+   * @param method - function / method to be executed by validator.
+   * @param baseConfig - baseConfig needed to properly connect to ledger
+
+   */
+  public sendAsyncRequest(
+    method: Record<string, unknown>,
+    args: any,
+    baseConfig?: IrohaBaseConfig,
+  ): void {
+    this.log.debug(`inside: sendAsyncRequest()`);
+    this.log.debug(`baseConfig=${baseConfig}`);
+    this.log.debug(`methodName=${method.methodName}`);
+    this.log.debug(`args=${args}`);
+
+    if (baseConfig === undefined || baseConfig === {}) {
+      throw new RuntimeError("baseConfig object must exist and not be empty");
+    }
+
+    if (
+      baseConfig.privKey === undefined ||
+      baseConfig.creatorAccountId === undefined ||
+      baseConfig.irohaHost === undefined ||
+      baseConfig.irohaPort === undefined ||
+      baseConfig.quorum === undefined ||
+      baseConfig.timeoutLimit === undefined
+    ) {
+      throw new RuntimeError("Some fields in baseConfig are undefined");
+    }
+
+    if (method.methodName === undefined || method.methodName === "") {
+      throw new RuntimeError("methodName parameter must be specified");
+    }
+
+    const socket: Socket = io(this.wsApiHost, { path: this.wsApiPath });
+    const asyncRequestData = {
+      baseConfig: baseConfig,
+      methodName: method.methodName,
+      args: args,
+    };
+
+    this.log.debug("requestData:", asyncRequestData);
+
+    try {
+      socket.emit(IrohaSocketSessionEvent.SendAsyncRequest, asyncRequestData);
+    } catch (err) {
+      this.log.error("Exception in: sendAsyncRequest(): ", err);
+      throw err;
+    }
+  }
+
+  /**
+   * Sends request to be executed on the ledger, watches and reports any error and the response from a ledger.
+   * @param args - arguments.
+   * @param method - function / method to be executed by validator.
+   * @param baseConfig - baseConfig needed to properly connect to ledger
+   * @returns Promise that will resolve with response from the ledger, or reject when error occurred.
+   */
+  public sendSyncRequest(
+    method: Record<string, unknown>,
+    args: any,
+    baseConfig?: IrohaBaseConfig,
+  ): Promise<any> {
+    this.log.debug(`inside: sendSyncRequest()`);
+    this.log.debug(`baseConfig=${baseConfig}`);
+    this.log.debug(`method=${method}`);
+    this.log.debug(`args=${args}`);
+
+    if (baseConfig === undefined || baseConfig === {}) {
+      throw new RuntimeError("baseConfig object must exist and not be empty");
+    }
+
+    if (
```

Comment:
@outSH Same question as above in the other method.

Reply:
@petermetz done

Reply:
@outSH Nice, thanks!